app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -159,81 +136,16 @@ fun ProfileScreen(viewModel: ProfileViewModel) {
 			onDismiss = { isDeleteAccountDialogVisible = false },
 		)
 	}
-	if (sheetState.isVisible) {
+	if (showProfileUpdateBottomSheet) {
 		userName = viewModel.userName.value
-		ModalBottomSheet(
-			modifier = Modifier
-				.padding(bottom = 56.dp),
-			sheetState = sheetState,
-			onDismissRequest = {
-				coroutineScope.launch {
-					sheetState.hide()
-					keyboardController?.hide()
-				}
-			},
-		) {
-			Row(horizontalArrangement = Arrangement.SpaceAround) {
-				CenterAlignedTopAppBar(
-					navigationIcon = {
-						IconButton(
-							onClick = {
-								coroutineScope.launch {
-									sheetState.hide()
-									keyboardController?.hide()
-								}
-							},
-						) {
-							Icon(Icons.Rounded.Close, contentDescription = null)
-						}
-					},
-					title = { Text(stringResource(id = R.string.EditUserData)) },
-					actions = {
-						IconButton(
-							onClick = {
-								viewModel.updateName(userName)
-								coroutineScope.launch {
-									sheetState.hide()
-									keyboardController?.hide()
-								}
-							},
-							enabled = viewModel.validatingName(userName).isValid(),
-						) {
-							Icon(Icons.Rounded.Check, contentDescription = null)
-						}
-					},
-				)
-			}
-			Column(
-				modifier = Modifier
-					.padding(start = 20.dp, end = 20.dp, bottom = 42.dp)
-					.fillMaxWidth(),
-			) {
-				TextField(
-					modifier = Modifier
-						.fillMaxWidth(),
-					value = userName,
-					onValueChange = {
-						userName = it
-					},
-					supportingText = {
-						if (!viewModel.validatingName(userName).isValid()) {
-							Text(
-								modifier = Modifier
-									.fillMaxWidth(),
-								text = stringResource(id = viewModel.validatingName(userName).errorMessageId),
-							)
-						}
-						Text(
-							text = "${userName.length} / 20",
-							modifier = Modifier.fillMaxWidth(),
-							textAlign = TextAlign.End,
-						)
-					},
-					isError = !viewModel.validatingName(userName).isValid(),
-					label = { Text(stringResource(id = R.string.UserName)) },
-				)
-			}
-		}
+		ProfileUpdateBottomSheet(
+			userName = userName,
+			updateUserName = { userName = it },
+			updateName = { viewModel.updateName(it) },
+			updateVisibility = { showProfileUpdateBottomSheet = it },
+			validateName = { viewModel.validatingName(it).isValid() },
```

Comment:
Of course `viewModel.validatingName(it).isValid()` it shouldn't be called in such way. What is the point of using view model here? 
Just a note, I see that it was used in that way previously

Reply:
As I can see `viewModel` is used only to separate logic. To not use `TextValidator` inside `ProfileUpdateBottomSheet`

Reply:
But it doesn't separate too much. `ValidationStatus` is returned from `viewModel::validatingName` but it shouldn't. As previously, it's a separate topic to not block this PR.